src/state/opamRepositoryState.ml

```diff
@@ -193,36 +196,56 @@ let load lock_kind gt =
         ) in
         Hashtbl.add repos_tmp name tmp
     ) repositories;
-  let make_rt repos_definitions opams =
+  let make_rt repos_definitions opams repos_sys_available_pkgs =
     let rt = {
       repos_global = (gt :> unlocked global_state);
       repos_lock = lock;
       repos_tmp;
       repositories;
       repos_definitions;
       repo_opams = opams;
+      repos_sys_available_pkgs
     } in
     OpamStd.Sys.at_exit (fun () -> cleanup rt);
     rt
   in
   match Cache.load gt.root with
-  | Some (repofiles, opams) ->
+  | Some (repofiles, opams, sys_available_pkgs) ->
     log "Cache found";
-    make_rt repofiles opams
+    make_rt repofiles opams sys_available_pkgs
   | None ->
     log "No cache found";
     OpamFilename.with_flock_upgrade `Lock_read lock @@ fun _ ->
-    let repofiles, opams =
-      OpamRepositoryName.Map.fold (fun name url (defs, opams) ->
+    let repofiles, opams, repo_depexts =
+      OpamRepositoryName.Map.fold (fun name url (defs, opams, all_depexts) ->
           let repo = mk_repo name url in
           let repo_def, repo_opams =
             load_repo repo (get_root_raw gt.root repos_tmp name)
           in
+          let repo_depexts = OpamFileTools.extract_depexts_map repo_opams
+              ~env:(OpamPackageVar.resolve_global gt)
+          in
+          let repo_depexts =
+            OpamSysPkg.Set.Op.(repo_depexts ++ all_depexts)
+          in
           OpamRepositoryName.Map.add name repo_def defs,
-          OpamRepositoryName.Map.add name repo_opams opams)
-        repos_map (OpamRepositoryName.Map.empty, OpamRepositoryName.Map.empty)
+          OpamRepositoryName.Map.add name repo_opams opams,
+          repo_depexts)
+        repos_map (OpamRepositoryName.Map.empty,
+                   OpamRepositoryName.Map.empty,
+                   OpamSysPkg.Set.empty)
+    in
+    let repos_sys_available_pkgs =
+    try
+      OpamSysInteract.available_packages ~env:gt.global_variables
+        gt.config repo_depexts
+    with Failure msg ->
+      OpamConsole.note "%s\nYou can disable this check using 'opam \
+                        option --global depext=false'"
+        msg;
+      Suppose_available
```

Comment:
For consistency, it should be an empty map

Reply:
Added a `No_depexts` variant instead